lib/SPIRV/libSPIRV/SPIRVError.h

```diff
@@ -109,12 +110,33 @@ class SPIRVErrorLog {
                   const std::string &DetailedMsg = "",
                   const char *CondString = nullptr,
                   const char *FileName = nullptr, unsigned LineNumber = 0);
+  // Check if Condition is satisfied and set ErrCode and DetailedMsg with Inst
+  // text representation if not. Returns true if no error.
+  bool checkError(bool Condition, SPIRVErrorCode ErrCode,
+                  llvm::Instruction *Inst, const std::string &DetailedMsg = "",
+                  const char *CondString = nullptr,
+                  const char *FileName = nullptr, unsigned LineNumber = 0);
 
 protected:
   SPIRVErrorCode ErrorCode;
   std::string ErrorMsg;
 };
 
+inline bool SPIRVErrorLog::checkError(bool Cond, SPIRVErrorCode ErrCode,
+                                      llvm::Instruction *Inst,
+                                      const std::string &Msg,
+                                      const char *CondString,
+                                      const char *FileName, unsigned LineNo) {
+  if (Cond)
+    return Cond;
+  // Do not overwrite previous failure.
+  if (ErrorCode != SPIRVEC_Success)
+    return Cond;
```

Comment:
Is this not already handled by the `checkError` function that you're calling below?

Reply:
Yes, it is, but the idea is that we should not call `toString` method earlier than it's actually needed. So, we need these 2 checks before the call. 
Maybe it can be done better to fully avoid code duplication, but I thought that such approach will result in more code and will worsen readability.

Reply:
Another way to solve the problem might be to move checking of the condition out of this function, so instead of 
```
      getErrorLog().checkError(
          DestAddrSpace != SPIRAS_Constant, SPIRVEC_InvalidModule, U,
          "Casts from generic address space to constant are illegal\n");
```
we would do something like:
```
      if (DestAddrSpace != SPIRAS_Constant)
        getErrorLog().reportErrorForInstruction(SPIRVEC_InvalidModule, U,
          "Casts from generic address space to constant are illegal\n");
```

Reply:
> Yes, it is, but the idea is that we should not call toString method earlier than it's actually needed.

Right, I see your point.  In that case it's worth adding a comment that you're doing the `Cond` check first for speed purposes.  Otherwise someone might simplify the function and remove it again one day.